src/sc.geometry.js

```diff
@@ -1,16 +1,30 @@
 'use strict';
 
-import spatialFeature from './sc.spatialfeature';
+import { spatialFeature } from './sc.spatialfeature';
 import * as _ from 'lodash';
 
-export default function(gj, storeId, layerId, spatialfeature) {
-  var scGeometry = {};
-  Object.setPrototypeOf(scGeometry, spatialFeature(storeId, layerId, spatialfeature));
+export function geometry(storeId, layerId, gj) {
+  var scGeometry = Object.create(spatialFeature(storeId, layerId, gj.properties));
+
+  var baseFeature = Object.getPrototypeOf(scGeometry).serialize();
+  scGeometry = _.defaults(gj, baseFeature);
+
+  if (scGeometry.geometry === undefined) {
+    scGeometry.geometry = null;
+  }
```

Comment:
Should we just validate the whole thing to make sure coordinates are correct and everything or is that overkill. Could use https://github.com/mapbox/geojsonhint